### PR TITLE
Prepare to move ARM types to subpackages

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Moves ARM types to subpackages, subject to a deny list that maintains the status quo.

We can remove groups from the deny list gradually, allowing a step-by-step migration to the new structure.

**Special notes for your reviewer**:

The count of changed lines looks large, but is almost entirely due to testing with the `containerservice` group. I can take that out if that's preferable.

**How does this PR make you feel**:
gif

### CODE REVIEW

**Comment by matthchr** [comment]
We think that the problem with some of the tests is this test-specific function:
`func CreateFileDefinition(definitions ...astmodel.TypeDefinition) *astmodel.FileDefinition {`

It needs to take a collection of packages, so that the codeGenContext has information about the imported packages too. Otherwise it can't find the arm types and fails.

**Comment by matthchr** [comment]
I also merged + pushed, we may need to re-record the golden file tests after my changes.

File: v2/pkg/genruntime/secrets/collector.go [hunk 1]
```diff
@@ -115,11 +116,12 @@ func (c *Collector) Values() ([]*v1.Secret, error) {
...
-	sort.Slice(result, func(i, j int) bool {
-		left := result[i]
-		right := result[j]
+	slices.SortFunc(result, func(left, right *v1.Secret) int {
+		if c := cmp.Compare(left.Namespace, right.Namespace); c != 0 {
+			return c
+		}
 
-		return left.Namespace < right.Namespace || (left.Namespace == right.Namespace && left.Name < right.Name)
+		return cmp.Compare(left.Name, right.Name)
```
Comment (matthchr) [inline v2/pkg/genruntime/secrets/collector.go:124]: Curious what the reason for the sortFunc change here is?

I'm not understanding what the difference is between what we had and what's here now, in terms of actual behavior.

File: v2/tools/generator/internal/astmodel/internal_type_name.go [hunk 2]
```diff
@@ -186,6 +186,10 @@ func (tn InternalTypeName) IsStatus() bool {
 
 // IsARMType returns true if the TypeName identifies an ARM specific type, false otherwise.
 func (tn InternalTypeName) IsARMType() bool {
+	if IsARMPackageReference(tn.InternalPackageReference()) {
+		return true
+	}
+
 	return strings.HasSuffix(tn.Name(), ARMSuffix)
```
Comment (matthchr) [inline v2/tools/generator/internal/astmodel/internal_type_name.go:193]: Can this second bit be removed if we migrate all types to the new format?

File: v2/tools/generator/internal/astmodel/type_name.go [hunk 3]
```diff
@@ -18,9 +20,58 @@ const (
...
+	"insights",
+	"keyvault",
+	"kubernetesconfiguration",
+	"machinelearningservices",
+	"managedidentity",
+	"monitor",
+	"network",
+	"network.frontdoor",
+	"operationalinsights",
+	//"person",
```
Comment (matthchr) [inline v2/tools/generator/internal/astmodel/type_name.go:56]: Intentionally here?

**Review by matthchr — APPROVED** [review]
LGTM, few minor questions

File: v2/pkg/genruntime/secrets/collector.go (hunk 1 above)
Comment (theunrepentantgeek) [reply v2/pkg/genruntime/secrets/collector.go:124]: I think we've both changed this. You added the ordering by namespace for packages with identical names; I updated it to use the new API. I'll update it.

File: v2/tools/generator/internal/astmodel/internal_type_name.go (hunk 2 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/internal_type_name.go:193]: Once we've moved every group, I'll remove both this and the deny list.

File: v2/tools/generator/internal/astmodel/type_name.go (hunk 3 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/type_name.go:56]: No, will remove.

**Pull request merged**